nixos/modules/services/x11/desktop-managers/default.nix

```diff
@@ -91,9 +99,15 @@ in
         example = "none";
         description = "Default desktop manager loaded if none have been chosen.";
         apply = defaultDM:
+          let
+            extraNames = concatMap (extraPackage: concatMap (d: builtins.match "(.*).desktop" d)
+              (builtins.attrNames (builtins.readDir "${extraPackage}/share/xsessions")))
```

Comment:
Build during evaluation is frowned upon for performance reasons, we will probably need to switch to a checking in later phase.